recipes-nemomobile/sensorfw/sensorfw-hybris-hal-plugins_git.bb

```diff
@@ -0,0 +1,18 @@
+require sensorfw.inc
+
+EXTRA_QMAKEVARS_PRE = "CONFIG+=autohybris"
+
+do_install:append() {
+    # Remove sensorfw core elements.
```

Comment:
How about
```
rm -rvf ${D}/!(usr) \
        ${D}/usr/!(lib) \
        ${D}/usr/lib/!(sensord-qt5)
```
? Maybe there are even more clever bash tricks but that's where my bash-fu stops :)

Reply:
This would be awesome if it worked :wink: 

Using any kind of pattern matching features appears to make it fail to build the package:

```
| DEBUG: Python function extend_recipe_sysroot finished
| DEBUG: Executing shell function do_install
| /home/darrel/AsteroidOS/asteroid-20220512/build/tmp-glibc/work/armv7vehf-neon-oe-linux-gnueabi/sensorfw-hybris-hal-plugins/+gitAUTOINC+2e53901599-r1/temp/run.do_install.3000954: line 166: syntax error near unexpected token `('
| WARNING: /home/darrel/AsteroidOS/asteroid-20220512/build/tmp-glibc/work/armv7vehf-neon-oe-linux-gnueabi/sensorfw-hybris-hal-plugins/+gitAUTOINC+2e53901599-r1/temp/run.do_install.3000954:152 exit 2 from 'export systemd_user_unitdir="/usr/lib/systemd/user"'
```

From what I can tell, pattern matching this way is a feature exclusive to bash. I could not find which command interpreter is used by bitbake.

A potential solution I found is shown here: https://unix.stackexchange.com/questions/670719/pattern-matching-in-if-block-in-alpines-sh
But in my opinion this will never look as neat as the one you showed.

So it looks like the current approach of explicitly deleting the core files might be the best way? Or do you know if I'm missing something here?

Reply:
Found a solution using `find`.